dlt/sources/helpers/rest_client/client.py

```diff
@@ -57,7 +59,7 @@ class RESTClient:
     Args:
         base_url (str): The base URL of the API to make requests to.
         headers (Optional[Dict[str, str]]): Default headers to include in all requests.
-        auth (Optional[AuthConfigBase]): Authentication configuration for all requests.
+        auth (Optional[AuthBase]): Authentication configuration for all requests.
```

Comment:
Will it work with out existing authenticators from `dlt.sources.helpers.rest_client.auth` which derive from `AuthConfigBase`? E.g., from our example code:

```py
from dlt.sources.helpers.rest_client import RESTClient
from dlt.sources.helpers.rest_client.auth import BearerTokenAuth

client = RESTClient(
    base_url="https://api.example.com",
    auth=BearerTokenAuth(token="your_access_token_here")  # type: ignore
)

for page in client.paginate("/protected/resource"):
    print(page)
```

Shouldn't it be a union of `AuthConfigBase` and `AuthBase` instead?

Reply:
this is due to linting problems that I merged from #1398 so I reverted it and we continue this review in #1413